subprojects/core/src/main/java/org/gradle/api/internal/plugins/DefaultPluginManager.java

```diff
@@ -307,6 +309,17 @@ public String getTargetPath() {
         public String getBuildPath() {
             return targetIdentifier.getBuildPath();
         }
+
+        @Override
+        public Object getCustomOperationTraceSerializableModel() {
+            Map<String, Object> map = new HashMap<String, Object>();
```

Comment:
Same comment, an immutable map would probably make sense, depending on the performance impact.

Reply:
Reverted back. Some values can be null, which is not permitted by `ImmutableMap`